### DISCUSSION TITLE
temporary workspace for application return to previous workspace on close

### DISCUSSION THREAD

**Discussion by tolson-vkn** [i3/i3#4707] [Q&A] [answered]
Hi. I've been looking to optimize my workspace a bit, I couldn't find this in the user guide.

I wish to have the following workflow:
* I exist on workspace X
* I want to change my audio settings so I hit mod+m and it launches an "Audio" workspace and launches pulsemixer
* I change my audio in this CLI app do whatever I need to
* I hit mod+Shift+Q and it closes the app AND -> Returns me to the previous workspace X (and thus close the temporary workspace)

I expect to need to run this workflow several times a day, so to be clear from the post I did manage to find, I am not intending this for startup layout configuration.

I have it "half" working with something like:

```
bindsym $mod+m exec --no-startup-id i3-msg 'workspace Audio'; exec alacritty --command pulsemixer;
```

This definitely works to launch pulsemixer but I have to close the app and manually select the workspace

It would be great if I had something like this is pseudo code:

```
bindsym $mod+m exec --no-startup-id i3-msg 'workspace Audio'; exec alacritty --command pulsemixer; #Wait for pulsemixer to close; #Navigate back to last workspace
```

I'd be happy with doing this in a completely different way, perhaps even with a bash script.

I've been banging my head trying to figure this out. Thanks for the help.

**Comment by Airblader** [comment, marked as answer]
> I'd be happy with doing this in a completely different way, perhaps even with a bash script.

There's probably no good way around that. You can use the IPC i3 offers to listen for events, and then e.g. if a window closes, the current workspace is "Audio" and that workspace is (now) empty, switch away from it.

**Reply by tolson-vkn** [reply]
Ok thank you. I was hoping for an easy answer and I had just missed something obvious. 

I've never used i3 IPC but will take a look based on your suggestion. I'll keep this open until tonight when I can try a POC. Where I will either have a solution or take it under advisement. Thanks for your assistance.
